cupy/cuda/memory.pyx

```diff
@@ -78,7 +78,7 @@ cdef class MemoryPointer:
         ptr (int): Pointer to the place within the buffer.
     """
 
-    def __init__(self, Memory mem, Py_ssize_t offset):
+    def __init__(self, mem, Py_ssize_t offset):
```

Comment:
We do not have to remove the type of `mem` if we make `CudaNdarrayMemory` a subclass of `Memory`.